src/plugins/lexer/types/mod.ts

```diff
@@ -0,0 +1,46 @@
+export type Trace = {
+  line: number;
+  column: [number, number];
+};
+export const enum Mode {
+  Token = 1,
+  String = 2,
+  Compiler = 3,
+}
+
+export type Token = {
+  type:
```

Comment:
The choice of lowercase strings was mainly preference, as the string values aren't planned to be in the final lexer/parser interactions.

On the discussion of not choosing numerical values, it's a topic of more discussion, as I didn't want to end up standardizing specific token identifiers as order numbers from 1 without a formal system. Mode was a more straightforward choice of numbering, as the data is local to the lexer; lexical token values on the other hand are sent to the parser, which may or may not be swapped by users later on as MTL becomes more modular.

Reply:
This is intelligent -- I totally agree. Since tokens can & probably will be used by target modules, we should be purposeful in their design. I don't mind the 1xx - 2xx - etc approach, however, could there ever be a situation where there are more than 99 tokens in one category? I don't think this would exist, but we want to be careful & not create a problem if something arises down the line. 

How do other languages accomplish this goal -- or do they not use numerics at all? Definitely worth diving deeper into this.

Reply:
Fixed in next commit :+1: